### DISCUSSION TITLE
升级Ng14

### DISCUSSION THREAD

**Discussion by wszgrcy** [wszgrcy/angular-miniprogram#36] [Announcements]
- 上周末尝试升级了下,部分接口类型没有了,不过可以有代替
- 比较坑爹的是rxjs angular-cli和angular不是一个版本的,所以还需要再独立一个包,将cli相关操作单独封装....这个比较耗时...

**Comment by 2286721642** [comment]
rxjs6 -> 7，老api还能用，能慢慢过渡到新版本

**Reply by wszgrcy** [reply]
能用但是不兼容.因为 `angular-cli`依赖的是内部的rxjs,外部的改成了rxjs 7 然后调用的时候添加的时候管道的类型(rxjs7)和默认的(rxjs6)不兼容...刚开始想都设置any对付下,但是后面发现太多了.不如抽出来

**Comment by 2286721642** [comment]
我写好了一个装饰器，怎么添加到这个项目里。放在哪个文件夹？这个装饰器是用于组件生命周期，可以直接在component.ts里写onReady onShow onReachBottom等，都能触发

**Reply by 2286721642** [reply]
<img width="739" alt="image" src="https://user-images.githubusercontent.com/43333114/176379613-acfe6748-af7d-4fc0-b2df-92ce8542faa9.png">

**Reply by wszgrcy** [reply]
目前此项目仅为实现小程序最核心的逻辑,对于一些优化开发体验的暂时没想到,您可以先自己建立一个仓库,等这次ng14升级完后我在考虑下如何划分(因为这次升级也会有一些包的分离,所以暂时没有这方面的打算),不过还是感谢支持,看来开发优先级要提前了......

**Comment by wszgrcy** [comment]
已升级完成
